src/Modules/Data/Commands.Data/GetPowerBITable.cs

```diff
@@ -0,0 +1,116 @@
+﻿/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+using System.Text;
+using Microsoft.PowerBI.Common.Abstractions;
+using Microsoft.PowerBI.Common.Abstractions.Interfaces;
+using Microsoft.PowerBI.Common.Api.Datasets;
+using Microsoft.PowerBI.Common.Api.Workspaces;
+using Microsoft.PowerBI.Common.Client;
+
+namespace Microsoft.PowerBI.Commands.Data
+{
+    [Cmdlet(CmdletVerb, CmdletName, DefaultParameterSetName = DatasetIdSetName)]
+    [OutputType(typeof(IEnumerable<Table>))]
+    public class GetPowerBITable : PowerBIClientCmdlet, IUserFirstSkip
```

Comment:
I think you should implement IUserScope and throw a NotImplementedException when Organization scope is specified in BeginProcessing. See RestorePowerBIWorkspace.cs for an example of this. Make Individual the default value for this optional parameter.

Including the Scope parameter for this cmdlet makes it clear that the entities returned are only for the individual user and that getting entities across the tenant will come at a future date.

Reply:
Okay but tables doesn't belong to either on Individual nor organization but to a dataset. That's why I exclude them. Please let me know if we still want to add it.

Reply:
The main point about Scope is the user's permission to access Power BI entities.

The user APIs (or cmdlet with Individual Scope) should only return tables if the user owns the specified dataset.
If the calling user is a Power BI tenant admin, the admin APIs (or cmdlet with Organization Scope) should return tables under any specified dataset, otherwise the person will get 403 Forbidden.

You can see examples of the difference between admin and user for the Dashboards APIs:
https://docs.microsoft.com/en-us/rest/api/power-bi/admin
https://docs.microsoft.com/en-us/rest/api/power-bi/dashboards
Compare GetDashboardsAsAdmin to Get Dashboards.

Please let me know if you have any questions. I think including Scope (Individual only) makes sense for this cmdlet even if the admin API doesn't exist yet.

Reply:
Understood. Added Scope and corresponding test